File: src/main/java/co/herod/adbwrapper/util/ImageUtil.java
package co.herod.adbwrapper.util;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

public class ImageUtil {

    public static void saveBufferedImage(final BufferedImage bufferedImage, final String pathname) throws IOException {
        ImageIO.write(bufferedImage, "PNG", FileUtil.getFile(pathname));
    }

    public static BufferedImage readImage(final File filePath) throws IOException {
        return ImageIO.read(filePath);
    }

    public static BufferedImage cropImage(final File filePath, final int x, final int y, final int w, final int h) throws IOException {
        return readImage(filePath).getSubimage(x, y, w, h);
    }
}

File: adb-kotlin/src/main/kotlin/co/herod/adbwrapper/util/ImageUtil.kt
/*
 * Copyright (c) 2018. Herod
 */

package co.herod.adbwrapper.util

import java.awt.image.BufferedImage
import java.io.File
import java.io.IOException
import javax.imageio.ImageIO

object ImageUtil {

    @Throws(IOException::class)
    fun saveBufferedImage(bufferedImage: BufferedImage, pathname: String) {
        ImageIO.write(bufferedImage, "PNG", FileUtil.getFile(pathname))
    }

    @Throws(IOException::class)
    private fun readImage(filePath: File): BufferedImage {
        return ImageIO.read(filePath)
    }

    @Throws(IOException::class)
    fun cropImage(filePath: File, x: Int?, y: Int?, w: Int, h: Int): BufferedImage? {
        return x?.let { y?.let { it1 -> readImage(filePath).getSubimage(it, it1, w, h) } }
    }
}
